crates/bevy_utils/src/parallel_queue.rs

```diff
@@ -21,6 +20,27 @@ impl<T: Send> Parallel<T> {
     pub fn clear(&mut self) {
         self.locals.clear();
     }
+
+    /// Retrieves the thread-local value for the current thread and runs `f` on it.
+    ///
+    /// If there is no thread-local value, it will be initialized to the result
+    /// of `create`.
+    pub fn scope_or<R>(&self, create: impl FnOnce() -> T, f: impl FnOnce(&mut T) -> R) -> R {
+        let mut cell = self.locals.get_or(|| RefCell::new(create())).borrow_mut();
+        let ret = f(cell.deref_mut());
+        ret
```

Comment:
I realize this was copied from existing code, but a few nits: 

- I don't think you need the locals here
- The first line might be simpler if you call `borrow_local_mut_or`
- I think `&mut cell` is more idiomatic than `cell.deref_mut()`

```suggestion
        f(&mut self.borrow_local_mut_or(create))
```